### PR TITLE
Fix alias analysis

### LINKED ISSUE #163: FusionPersistentSoftmaxLocalShared failure
On my Titan RTX, the FusionPersistentSoftmaxLocalShared_CUDA test fails:

```
C++ exception with description "(dynamic_smem_size) < (available_dynamic_smem_without_reconfiguration + additional_dynamic_smem_available_through_reconfiguration) INTERNAL ASSERT FAILED at "/home/nmaruyama/pytorch/debug3/nvfuser/csrc/executor.cpp":910, please report a bug to PyTorch. The total shared memory allocation is larger than available memory. Dynamic size: 66048. Available size: 49136. Configured smem size: 49152. Device limit size: 65536
```

It seems this issue started to happen at PR #148. Unclear why the PR could affect shared memory usage.

### PR DESCRIPTION
**Pull request by naoyam**
Set was replaced with LoadStoreOp, which was not recognized as UnaryOp, and thus the alias analysis failed to detect safe aliasing

Fixes #163

### CODE REVIEW

File: csrc/ir_utils.cpp [hunk 1]
```diff
@@ -572,6 +572,11 @@ bool isReductionTvOp(const Expr* expr) {
   return ir_utils::isTvOp(expr) && isReductionOp(expr);
 }
 
+bool isPointwiseTvOp(const Expr* expr) {
+  return isTvOp(expr) &&
+      expr->isOneOf<UnaryOp, BinaryOp, TernaryOp, LoadStoreOp>();
```
[the remark this answers is not in the document]
Comment (zasdfgbnm) [reply csrc/ir_utils.cpp:577]:
```
T2 = transpose(T1)
T3 = sin(T2)
```
For the example above, I think it is OK to share T3 with T1.

**Review by zasdfgbnm — APPROVED** [review]

File: csrc/ir_utils.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (naoyam) [reply csrc/ir_utils.cpp:577]: That's a good point. Is it really safe? Suppose `t1 = transpose(t0)`, and both of them are on shared memory, code like below seemed to be generated:

```
__shared__ t0[N] = ...;
auto& t1 = t0;
t1[i * k + j] = t0[j *k + i];
```

This isn't valid an it's not really pointwise, is it?

[the remark this answers is not in the document]
Comment (zasdfgbnm) [reply csrc/ir_utils.cpp:577]: Wait what, is t1 sharing with t0? I think we previously disabled it. https://github.com/csarofeen/pytorch/pull/2490

[the remark this answers is not in the document]
Comment (naoyam) [reply csrc/ir_utils.cpp:577]: That is right. But should we call it a pointwise op?

[the remark this answers is not in the document]
Comment (naoyam) [reply csrc/ir_utils.cpp]: Updated.

**Comment by naoyam** [comment]
!build

**Pull request merged**